source/CommonJobs/CommonJobs.Application.EvalForm/EmployeeSearching/EmployeeToEvaluate_Search.cs

```diff
@@ -0,0 +1,96 @@
+﻿using CommonJobs.Domain.Evaluations;
+using Raven.Abstractions.Indexing;
+using Raven.Client.Indexes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace CommonJobs.Application.EvalForm.EmployeeSearching
+{
+    public class EmployeeToEvaluate_Search : AbstractMultiMapIndexCreationTask<EmployeeToEvaluate_Search.Projection>
+    {
+
+        public class Projection
+        {
+            public string Id { get; set; }
+            public string UserName { get; set; }
+            public string FullName { get; set; }
+            public string CurrentPosition { get; set; }
+            public string Seniority { get; set; }
+            public string Period { get; set; }
+            public string[] Evaluators { get; set; }
+            public string Responsible { get; set; }
+            public bool AutoEvaluationDone { get; set; }
+            public bool ResponsibleEvaluationDone { get; set; }
+            public bool CompanyEvaluationDone { get; set; }
+            public bool OpenToDevolution { get; set; }
+            public bool Finished { get; set; }
+        }
+
+        public EmployeeToEvaluate_Search()
+        {
+            AddMap<EmployeeEvaluation>(evaluations =>
+                from evaluation in evaluations
+                select new
+                {
+                    Id = evaluation.Id,
+                    UserName = evaluation.UserName,
+                    FullName = evaluation.FullName,
+                    CurrentPosition = evaluation.CurrentPosition,
+                    Seniority = evaluation.Seniority,
+                    Period = evaluation.Period,
+                    Evaluators = new dynamic[0],
+                    Responsible = evaluation.Responsible,
+                    AutoEvaluationDone = false,
+                    ResponsibleEvaluationDone = false,
+                    CompanyEvaluationDone = false,
+                    OpenToDevolution = evaluation.OpenToDevolution,
+                    Finished = evaluation.Finished
+                });
```

Comment:
Well, the nice thing about this approach is that you have the "state" of everything that has happened to the evaluation, so that's good.

However, if you convert this into an Enum and somehow update the enum (since that's part of the model), you may not be able to easily translate the enum into these properties... unless you keep track of all of them, in which case the Enum is just redundant, or a calculated value.

If you wish to send the enum to Raven, you can always convert it into a string and send it over, and then parse it when coming from the database. Ints are the usual approach but:

1. They're not readable from the DB, strings are.
2. It's easy to make a mistake if you change the enum. With ints, the values will take new meanings (which could be a big problem that you find late in the game), but with strings, if you forget to update the DB, you will find values that cannot be parsed right away, which is the worst scenario.